modules/runtime-info/src/main/java/org/opencastproject/runtimeinfo/rest/DocRestData.java

```diff
@@ -193,21 +193,16 @@ public void setAbstract(String abstractText) {
    * @return true if this path is valid, false otherwise
    */
   public static boolean isValidPath(String path) {
-    boolean valid = true;
```

Comment:
Maybe just

`
return (!isBlank(path) && path.matches("^/[\\w/{}.]+$"));
`
?

Reply:
No, the characters do not need to be escaped. Why would they need that?

And your return statement is not equivalent to the current functionality. That would be:

```java
return isNotBlank(path) 
  && (SLASH.equals(path)
      || (path.matches("^/[\\w/{}.]*$")
          && !path.endsWith("/")));
```

And I agreed to the previous author that not packing that in one line is easier to read.

Reply:
Though you could pack the last two conditions in one regexp:

```java
path.matches("^/[\\w/{}.]*[\\w}.]$");
```

Reply:
> No, the characters do not need to be escaped. Why would they need that?

The escaping idea was because ".", "{" and "}" are special characters and I've seen examples regex where they were escaped. Absolutely possible that this is not required. The online regex evaluator I've tried did also not require that, although it did not work with "\\w" but did with "\w" ;-)
-> If that works, fine for me.

What about 

`return isNotBlank() && path.matches("^/$|^[\\w/{}.]+[^/]$")`

That one does match "/", requires start with "/" and also does not match anything that ends with "/".

Anyway, I'm of course also fine with the if-statement approach although I'm not a big fan of decomposing expressions into multiple if-statements. That again is personal taste, though.

Should I accept as is or could I convince you?

;-)

Reply:
I think this discussion is becomming a good argument for avoiding complex regular expressions. `path.matches("^/$|^[\\w/{}.]+[^/]$")` would now accept probably unwanted characters as last character. `path.matches("^/$|^/[\\w/{}.]*[\\w}.]$");` would probably be the way to go then ;-P

I'll change it to this tomorrow:

```java
return isNotBlank() && path.matches("^/$|^/[\\w/{}.]*[\\w}.]$");
```

For the escaping: Java strings itself need escaping of `\`. Hence internally, `\\w` will become `\w`. Regular characters which have no special meaning don't need to be escaped. For example, in between `[` and `]` a `.` has no special meaning. It's just the character `.`.

Reply:
Done